akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -86,6 +82,53 @@ import akka.util.Helpers.toRootLowerCase
     (bytes(0) == GZIPInputStream.GZIP_MAGIC.toByte) &&
     (bytes(1) == (GZIPInputStream.GZIP_MAGIC >> 8).toByte)
   }
+
+  case class LZ4Meta(offset: Int, length: Int) {
+    import LZ4Meta._
+
+    def putInto(buffer: ByteBuffer): Unit = {
+      buffer.putInt(LZ4_MAGIC)
+      buffer.putInt(length)
+    }
+
+    def prependTo(bytes: Array[Byte]): Array[Byte] = {
+      val buffer = ByteBuffer.allocate(bytes.length + offset)
+      putInto(buffer)
+      buffer.put(bytes)
+      buffer.array()
+    }
+
+  }
+
+  object LZ4Meta {
+    val LZ4_MAGIC = 0x87d96df6 // The last 4 bytes of `printf akka | sha512sum`
+
+    def apply(bytes: Array[Byte]): LZ4Meta = {
+      LZ4Meta(8, bytes.length)
+    }
+
+    def get(buffer: ByteBuffer): OptionVal[LZ4Meta] = {
+      try {
+        if (buffer.getInt() != LZ4_MAGIC) {
+          OptionVal.None
+        } else {
+          OptionVal.Some(LZ4Meta(8, buffer.getInt()))
+        }
+      } catch {
+        case _: java.nio.BufferUnderflowException => OptionVal.None
```

Comment:
This would be if the buffer has less than four bytes, I guess that will be uncommon, but if there are such cases (it will be called for every message in fromBinary), creating a `BufferUnderFlow` will capture a stack trace which is expensive, so perhaps checking buffer length to avoid ever creating the exception would be worth it.